### PR TITLE
Add Microsoft.Bot.Solutions SDK UT

### PR DESCRIPTION
**Pull request by sidmore**
### Purpose
*What is the context of this pull request? Why is it being done?*
Add missing UT's

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: sdk/csharp/libraries/microsoft.bot.solutions/BotSettingsBase.cs [hunk 1]
```diff
@@ -13,6 +14,7 @@ namespace Microsoft.Bot.Solutions
     /// <summary>
     /// Base class representing the configuration for a bot.
     /// </summary>
+    [ExcludeFromCodeCoverageAttribute]
```
Comment (ryanisgrig) [inline sdk/csharp/libraries/microsoft.bot.solutions/BotSettingsBase.cs:17]: I didn't know about this, great!

**Review by ryanisgrig — APPROVED** [review]

**Review by taicchoumsft — APPROVED** [review]
:shipit:

File: sdk/csharp/libraries/microsoft.bot.solutions/BotSettingsBase.cs (hunk 1 above)
Comment (taicchoumsft) [inline sdk/csharp/libraries/microsoft.bot.solutions/BotSettingsBase.cs:17]: >Attribute [](start = 28, length = 9)

minor nit: can remove the  `attribute` part of the name.

File: sdk/csharp/tests/microsoft.bot.solutions.tests/Authentication/AuthenticationTests.cs [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+        {
+            var messageActivity = activity.AsMessageActivity();
+            Assert.IsNotNull(messageActivity);
+            Assert.AreEqual("Logged in.", messageActivity.Text);
+        })
+        .StartTestAsync();
+        }
+
+        [TestMethod]
+        [Obsolete]
```
Comment (taicchoumsft) [inline sdk/csharp/tests/microsoft.bot.solutions.tests/Authentication/AuthenticationTests.cs:187]: >Obsolete [](start = 9, length = 8)

little question: should we remove obsolete tests?

**Pull request merged**